sound/soc/sof/ipc4-loader.c

```diff
@@ -41,19 +41,25 @@ static size_t sof_ipc4_fw_parse_ext_man(struct snd_sof_dev *sdev)
 
 	ext_man_hdr = (struct sof_ext_manifest4_hdr *)fw->data;
 
+	/* Verify the magic number if the firmware image */
+	if (ext_man_hdr->id == SOF_EXT_MAN4_MAGIC_NUMBER)
+		ext_man_len = ext_man_hdr->len;
+	else
+		ext_man_len = 0;
```

Comment:
if ext_man_len is 0, why proceed with any of the parsing below?

Reply:
In theory there can be a firmware file w/o manifest, in that case we try to verify that the next header is OK to make sure that we don't discard otherwise valid image

Reply:
@ujfalusi not following you. If there's no extended manifest , the FW image is valid alright but the code below is meant for parsing the extended manifest no? Why do we need to do those?

Reply:
It is not for parsing the extended manifest which is skipped and is for normal ADSP manifest. But why not return invalid error ?

Reply:
If we want to be pragmatic, the non existence of a non mandatory ext_manifest section is not an error.
The first check is just complements the second one by failing the parsing in case we end up pointing to an invalid firmware section after the ext_manifest (be it a valid ext_manifest4 or non existent one or an incompatible manifest section).

Sidenote: I think we should rename the `parse_ext_manifest` ops to `parse_firmware_image` or something as we do parse the firmware section in case of an IPC4 image.

@ranj063, yes, we could just return right away if the magic is not found, but we need the modules to be parsed out in this phase (or fail).

Reply:
@RanderWang, if it is absolutely mandatory to have ext_manifest4, then I would be more than happy to return just an error as I did in the first version.

Reply:
Windows driver requires this ext part. If we want to use single fw for both windows and Linux, we have to include it. And current rimage generates ext manifest unconditionally.